runtime/nativeMain/src/kotlinx/benchmark/native/NativeExecutor.kt

```diff
@@ -82,11 +181,13 @@ class NativeExecutor(name: String, args: Array<out String>) : SuiteExecutor(name
     ): Double {
         val executeFunction = benchmark.function
         var counter = cycles
+        GC.collect()
         val startTime = getTimeNanos()
         while (counter-- > 0) {
             @Suppress("UNUSED_VARIABLE")
             val result = instance.executeFunction() // ignore result for now, but might need to consume it somehow
         }
+        GC.collect()
```

Comment:
Unconditional GC between each measurement iteration implies multiple things:

- Small micro-nano benchmarks may be dominated by this runtime call with undesired side-effects
- The amortized cost of allocations (especially TLAB-like) is simply not measured. E.g. it would be impossible to compare two JSON parsers that differentiate mostly in allocation patterns 
- We potentially mess up with CPU caches on each iteration. That's not really helpful for correct measurements and comparison, especially when comparing regular and cache-obvious algorithms

I strongly suggest making GC after each iteration optional and disabled by default

Reply:
I got your concerns, I can do several modes. We can even use fully turned off GC as the second mode. I really don't like an idea to. run benchmark with only auto GC calls made by runtime, because then measurements can differ a lot between different runs of benchmarks. Moreover, some users expect that the result of running same benchmark as separate application should be close to results of benchmarking with library, that's impossible with auto-called GC.

What do you think if we make 2 modes: one with calling GC after each iteration (behaviour similar to separate application) and without GC calls at all?

Reply:
IMO It's hard to actually have non-micro benchmarks without GC at all, but in my (JVM-ish) experience I mostly used to measure GC as amortized part of the execution. And in really rare cases (whether I want precision, a minimal amount of noise or do not care about memory pressure) I use `-prof gc` that collects garbage after each iteration.

I think it's ok to have three modes for GC: `none`, `auto` and `iteration` with `auto` as default